cpp/include/cudf/detail/aggregation/aggregation.cuh

```diff
@@ -261,7 +261,9 @@ struct update_target_element<
  */
 template <bool target_has_nulls = true, bool source_has_nulls = true>
 struct elementwise_aggregator {
-  template <typename Source, aggregation::Kind k>
+  template <typename Source,
+            aggregation::Kind k,
+            std::enable_if_t<cudf::is_relationally_comparable<Source, Source>()>* = nullptr>
```

Comment:
This isn't the right place for this specialization. `element_wise_aggregator` is used for more than relational comparison operations. This should be a partial specialization of `update_target_element` for relational aggregations like MIN/MAX.